### PR TITLE
New StringProperty 'Display Name' that allows to distinguish multiple…

### PR DESCRIPTION
**Pull request by NikolasE**
… instances. The name is so far only used for the trigger service, but could also be used for topics to set framerate. The service is now also relative to rviz so that rviz should be started with a fixed name

### CODE REVIEW

File: src/camera_display.cpp [hunk 1]
```diff
@@ -374,6 +379,50 @@ void CameraPub::updateFrameRate()
...
+                  "Could not create trigger. Make sure that display name is unique!");
+        return;
+    }
+
+    setStatus(StatusProperty::Ok, "Display name", "OK");
+}
+
+std::string CameraPub::findUnusedDisplayName()
+{
+    ros::NodeHandle nh_rviz("~"); /// get namespace of rviz
```
Comment (lucasw) [inline src/camera_display.cpp]: Does rviz::Display::update_nh_ work for this?

File: src/camera_display.cpp [hunk 2]
```diff
@@ -374,6 +379,50 @@ void CameraPub::updateFrameRate()
...
+    {
+        setStatus(StatusProperty::Warn, "Display name",
+                  "Could not create trigger. Make sure that display name is unique!");
+        return;
+    }
+
+    setStatus(StatusProperty::Ok, "Display name", "OK");
+}
+
+std::string CameraPub::findUnusedDisplayName()
```
Comment (lucasw) [inline src/camera_display.cpp]: I'd be curious if there is any existing code for this- `ros::init_options::AnonymousName` just generates a random number (haven't tried it though), roslaunch $(anon foo) creates a name like `foo_hostname_15435_769474121848175325-4`

File: src/camera_display.cpp [hunk 3]
```diff
@@ -374,6 +379,50 @@ void CameraPub::updateFrameRate()
...
+{
+    std::string name = name_property_->getStdString();
+    if (name.empty())
+    {
+        setStatus(StatusProperty::Warn, "Display name",
+                  "Should not be empty.");
+        return;
+    }
+
+    ros::NodeHandle nh_private_ = ros::NodeHandle("~"+name);
```
Comment (lucasw) [inline src/camera_display.cpp]: Without the forced `~` prefix the user would able to select whether they want to use the tilde in the string property, no tilde (which moves up one namespace?), or put a leading slash for an absolute namespace.

File: src/camera_display.cpp [hunk 4]
```diff
@@ -159,21 +159,23 @@ bool validateFloats(const sensor_msgs::CameraInfo& msg)
...
-  trigger_service_ = nh_.advertiseService("/rviz_camera_trigger",
-      &CameraPub::triggerCallback, this);
 
   topic_property_ = new RosTopicProperty("Image Topic", "",
       QString::fromStdString(ros::message_traits::datatype<sensor_msgs::Image>()),
       "sensor_msgs::Image topic to publish to.", this, SLOT(updateTopic()));
 
+  // std::string default_display_name = findUnusedDisplayName();
+  name_property_ = new StringProperty("Display name", "", //QString(default_display_name.c_str()
+      "Name fo this camera display", this, SLOT(updateDisplayName()));
```
Comment (lucasw) [inline src/camera_display.cpp]: fo -> for

File: src/camera_display.cpp (hunk 3 above)
Comment (NikolasE) [inline src/camera_display.cpp]: In this case, we should rename it to "display namespace" so that's more obvious that namespace rules apply.

Comment (lucasw) [inline src/camera_display.cpp]: That sounds good.

**Comment by lucasw** [comment]
I'm not seeing the camera trigger service here- one problem may be that it is only created after the namespace is set, but even after `updateDisplayNamespace` is run I still don't have that service.

**Comment by NikolasE** [comment]
Sorry, I shut down the service when the image was reconfigured but only set it up when configuring the namespace.

File: src/camera_display.cpp [hunk 5]
```diff
@@ -76,6 +76,11 @@ class VideoPublisher
   {
   }
 
+  bool is_active()
+  {
+      return not pub_.getTopic().empty();
```
Comment (lucasw) [inline src/camera_display.cpp]: Just use two spaces to indent (also a few more of these below).

**Comment by lucasw** [comment]
1. Run `catkin_make roslint_rviz_camera_stream` and fix all the warnings (roslint is available as a ros deb if you don't already have it).  Also run it again after the below steps just to make sure nothing new was introduced.

1. Resolve the conflicts with master (I believe this will be a rebase).

1. Squash all the commits for this pr.  (maybe this is more easily done before the above step and after)

If you do all that then I don't have to manually modify and merge, and if I merge through github it will properly show the pr as accepted.

**Pull request closed without merging**